private/client-api-test/src/client-s3-interface.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { S3ClientResolvedConfig } from "@aws-sdk/client-s3";
+
+import { initializeWithMaximalConfiguration, initializeWithMinimalConfiguration } from "./client-s3-interface";
+
+export type FIELD_INIT_TYPE = "resolvedByConfigResolver" | "resolvedOnlyIfProvided" | "neverResolved";
+
+const expectedResolvedFields: Record<keyof S3ClientResolvedConfig, FIELD_INIT_TYPE> = {
```

Comment:
Move `expectedResolvedFields` to it's own file, say `src/client-s3/expected/RESOLVED_FIELDS.ts`? Also, change the data structure to `Record<FIELD_RESOLVED_TYPE, Array<keyof S3ClientResolvedConfig>>` for ease of readability?

For example:
```js
export enum FIELD_RESOLVED_TYPE {
  CONFIG: "Config",
  CUSTOM: "Custom",
  NEVER: "Never"
}
```

```js
export const RESOLVED_FIELDS_S3 = {
   FIELD_RESOLVED_TYPE.CONFIG: ["requestHandler", "apiVersion" /** others */],
   FIELD_RESOLVED_TYPE.CUSTOM: ["customUserAgent" /** others */]
}
```

We can write generic fields later, like in `src/client/expected/RESOLVED_FIELDS.ts` which other clients (including S3) can extend from, and/or make these tests fully/partly automated during codegen.